Source/WebGPU/WGSL/Overload.cpp

```diff
@@ -363,10 +362,83 @@ bool OverloadResolver::unify(const AbstractValue& parameter, unsigned argumentVa
     return *resolvedValue == argumentValue;
 }
 
-void OverloadResolver::assign(TypeVariable variable, Type* type, Substitution::Inferred inferred)
+bool OverloadResolver::assign(TypeVariable variable, Type* type, Substitution::Inferred inferred)
 {
     logLn("assign ", variable, " => ", *type);
+    if (variable.constraints) {
+        auto* primitive = std::get_if<Types::Primitive>(type);
```

Comment:
we generally use `downcast<>()` in WebKit instead; is there a benefit to `get_if`?

Reply:
`get_if` is for variants, `downcast` is for subclass, and this type is a variant. @djg suggested moving to class and using `downcast` instead, but I haven't gotten around to it yet.